tests/v2/validation/v2prov/encryption_key_rotation_test.go

```diff
@@ -170,16 +184,46 @@ func (r *V2ProvEncryptionKeyRotationTestSuite) TestEncryptionKeyRotation() {
 	setEncryptSecret(r.T(), r.client, id)
 
 	prefix := "encryption-key-rotation-"
+
+	r.Run(prefix+"new-cluster", func() {
+		rotateEncryptionKeys(r.T(), r.client, id, 10*time.Minute)
+	})
+
+	if r.client.Flags.GetValue(environmentflag.Long) {
+		// create 10k secrets for stress test, takes ~30 minutes
+		createSecretsForCluster(r.T(), r.client, id, totalSecrets)
+
+		r.Run(prefix+"stress-test", func() {
+			rotateEncryptionKeys(r.T(), r.client, id, 1*time.Hour) // takes ~45 minutes for HA
+		})
+	}
+}
+
+func (r *V2ProvEncryptionKeyRotationTestSuite) TestEncryptionKeyRotationWithVaiEnabled() {
+	subSession := r.session.NewSession()
+	defer subSession.Cleanup()
+
+	id, err := clusters.GetV1ProvisioningClusterByName(r.client, r.clusterName)
+	require.NoError(r.T(), err)
+
+	setEncryptSecret(r.T(), r.client, id)
+	enableVai := true
+	var once sync.Once
+	err = vai.EnableVAI(r.client, &enableVai, &once)
+	require.NoError(r.T(), err)
+
+	prefix := "encryption-key-rotation-vai-enabled-"
```

Comment:
I think it would be worth adding the cluster type (k3s, rke2) to the prefix name

Also, are we going to test on all k8s versions, or just the latest? Either way I think it would also be worth adding the k8s version to the name of the test too, similar to how we do in provisioning tets / etcd snapshot restore tests if you need an example.

Reply:
I thought a bit about this. I agree that adding the cluster type (k3s, rke2) to the test prefix is a good idea, mad that change. but, for the Kubernetes version, this test is going to run against the latest version. Since there won’t be any variation across versions in this particular case, adding the version feels unnecessary and would the test names bulky without much value. If, in the future, we want to test against multiple k8s versions, I can adjust this accordingly. For now, I'll update the test names to reflect the cluster type and leave out the Kubernetes version.